cpp/tests/test_save_results.cpp

```diff
@@ -105,3 +113,188 @@ TEST(TestSaveResult, compareResultWithH5)
+        params.get<mio::osecir::SeverePerInfectedSymptoms>()[i]        = 0.2;
+        params.get<mio::osecir::CriticalPerSevere>()[i]                = 0.25;
+        params.get<mio::osecir::DeathsPerCritical>()[i]                = 0.3;
+    }
+
+    mio::ContactMatrixGroup& contact_matrix = params.get<mio::osecir::ContactPatterns>();
+    contact_matrix[0] = mio::ContactMatrix(Eigen::MatrixXd::Constant(num_groups, num_groups, fact * cont_freq));
+
+    mio::osecir::set_params_distributions_normal(model, t0, tmax, 0.2);
+
+    auto graph = mio::Graph<mio::osecir::Model, mio::MigrationParameters>();
+    graph.add_node(0, model);
+    graph.add_node(1, model);
+    graph.add_edge(0, 1, mio::MigrationParameters(Eigen::VectorXd::Constant(Eigen::Index(num_groups * 8), 1.0)));
+
+    auto num_runs = 3;
+    auto study    = mio::ParameterStudy<mio::osecir::Simulation<>>(graph, 0.0, 2.0, 0.5, num_runs);
+    auto results  = study.run([](auto&& g) {
+        return draw_sample(g);
+    });
+
+    TempFileRegister tmp_file_register;
+    std::string tmp_results_dir = tmp_file_register.get_unique_path();
+    ASSERT_THAT(mio::create_directory(tmp_results_dir), IsSuccess());
+
+    auto ensemble_results = std::vector<std::vector<mio::TimeSeries<double>>>{};
+    ensemble_results.reserve(size_t(num_runs));
+    auto ensemble_params = std::vector<std::vector<mio::osecir::Model>>{};
+    ensemble_params.reserve(size_t(num_runs));
+    auto save_result_status = mio::IOResult<void>(mio::success());
+    auto run_idx            = size_t(0);
+    study.run(
+        [](auto&& g) {
+            return draw_sample(g);
+        },
+        [&](auto results_graph) {
+            ensemble_results.push_back(mio::interpolate_simulation_result(results_graph));
+
+            ensemble_params.emplace_back();
+            ensemble_params.back().reserve(results_graph.nodes().size());
+            std::transform(results_graph.nodes().begin(), results_graph.nodes().end(),
+                           std::back_inserter(ensemble_params.back()), [](auto&& node) {
+                               return node.property.get_simulation().get_model();
+                           });
+
+            save_result_status = save_result_with_params(ensemble_results.back(), ensemble_params.back(), {0, 1},
+                                                         tmp_results_dir, run_idx);
+
+            ++run_idx;
+        });
+    ASSERT_TRUE(save_result_status);
+    auto results_from_file = mio::read_result(tmp_results_dir + "/run0/Result.h5");
+    ASSERT_TRUE(results_from_file);
+    auto result_from_file = results_from_file.value()[0];
+    EXPECT_EQ(ensemble_results.back().back().get_num_elements(), result_from_file.get_groups().get_num_elements());
+    EXPECT_EQ(ensemble_results.back().back().get_num_time_points(),
+              result_from_file.get_groups().get_num_time_points());
+}
+
+TEST(TestSaveResult, save_results)
```

Comment:
There is already a test called 'save_result'. You could name this one save_ensemble_results or sth like that